web/src/components/people/ExternalInvite.tsx

```diff
@@ -0,0 +1,114 @@
+import Button from "@/components/ui/Button";
+import { useEffect, useState } from "react";
+import { Input, Alert, AlertMessage, Label } from "../shared/exports";
+import styles from "./ExternalInvite.module.css";
+import { postUsersInvite } from "@/openapi";
+import Loading from "@/components/ui/Loading";
+import Dialog from "../ui/Dialog";
+
+export default function ExternalInvite() {
+  const [buttonDisabled, setButtonDisabled] = useState(false);
+  const [isDialogVisible, setDialogVisible] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [email, setEmail] = useState("");
+  const [showSuccessMessage, setShowSuccessMessage] = useState(false);
+  const [showErrorMessage, setShowErrorMessage] = useState(false);
+  const [errorMessage, setErrorMessage] = useState("");
+
+  function handleShowDialog() {
+    setDialogVisible(true);
+  }
+
+  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    try {
+      setButtonDisabled(true);
+      setIsLoading(true);
+      if (typeof email === "string") {
```

Comment:
what's this type check for? I don't think it can be anything other than a string(?)